src/pull.go

```diff
@@ -82,6 +88,72 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type rename struct {
+		name string
+		change *Change
+	}
+	clashesMap := map[string][]rename{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil { // Is that even possible? In what situation?
+			continue
+		}
+		clashesMap[clash.Path] = append(group, rename{"", clash})
+	}
+
+	for path, group := range clashesMap {
+		ext := filepath.Ext(path)
+		name := strings.TrimSuffix(path, ext)
+		n := 0
+
+		for i, r := range group[1:] {  // we can leave first item with original name
+			var newName string
+			for {
+				newName = fmt.Sprintf("%v_%d%v", name, n, ext)
+				n++
+
+				dupCheck, err := g.rem.FindByPath(newName)
```

Comment:
We may already have a file_0.jpg in our directory. I'm creating a list of renames here so that I could present it to user and confirm this is what he wants. I know it's a little racy because new file named file_0.jpg may appear in the meantime but this the best we can do, I guess.